docs/_docs/03-ug-mutable-dataset.md

```diff
@@ -87,7 +89,84 @@ hs.refreshIndex("empIndex", "incremental")
 ```
 
 ### Optimize Index
-TODO
+The fastest way to index newly added source data files and merge them into an existing index is by calling `refreshIndex`
+with the `"incremental"` mode. In this mode, each time refresh is called, it creates fresh index files for new data files
+and updates index metadata to include them in the index content.
+As these index files accumulate, they could affect query performance when the index is used.
+Once the index is leveraged for a query,
+the large number of these files could increase overall query time as more index files need to be accessed and potentially
+read to compute the query results.
+Hyperspace provides the `optimizeIndex` command to alleviate above issue by changing index files layout for an
```

Comment:
I think we need to show the status of index data files (e.g. number of files) and suggest to run optimizeIndex so that a user can easily determine and trigger it if needed. Could you create an issue for this?

Reply:
Sure, are suggesting to add a new API to Hyperspace which can be called directly by user and shows some stats on the index, including how "fragmented" it is?

Reply:
Yes we could show 
- general index stats
  - name, index data bytes,
  - all versions for index data files of (latest version?)
     - can be multiple in case optimizeIndex or incremental refresh)
- source data context - (we could merge with isValid (#22))
  - number of current files / appended files / deleted files 
  - show 3~5 full paths of appended or deleted fils 
- optimize context
  - how many optimize-ndeeded files and their size
  - recommend optimizeIndex or not?
- root paths of index source files 

I think we could do this with 
hs.getIndexStatus(IndexName) or just hs.optimizeIndex(IndexName, mode, dryRun = true) only for "optimize context"?
WDYT? @pirz

Reply:
@sezruby - Overall looks fine to me. I just think:
1) We should keep it as a separate command (like what you suggest: `getIndexStatus(ixName)` - or probably rename it with sth else) and we should not mix it with Optimize.
2) "recommend optimizeIndex or not?": I am not sure if it is a good idea to make a such a recommendation in the output of such an API. Normally, a "getStatus" type of API should be metadata read/summarization only. Making such a recommendation needs some (deeper) index content analysis. We need to check how much of overlap we have between new index file's buckets and the older ones. This does not necessarily need scanning index records, but can be done by extracting bucket-ids from file names and grouping them on that to see how many buckets with ">1 files" are there. But in general, "getStatus()" APIs and "recommend()" APIs are normally separated. 
Adding @rapoth - @apoorvedave1 - @imback82 to make suggestions